clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -635,46 +626,52 @@ RegisterBindingFlags HLSLFillRegisterBindingFlags(Sema &S, Decl *D) {
       } else
         r.Other = true;
     }
-  } else {
-    llvm_unreachable("unknown decl type");
   }
   return r;
 }
 
+int getRegisterTypeIndex(StringRef Slot) {
+  switch (Slot[0]) {
+  case 't':
```

Comment:
Good news!  It looks like DXC is case insensitive for register types: https://godbolt.org/z/e9Y71o9ch